Configurations/10-main.conf

```diff
@@ -965,10 +965,76 @@ sub vms_info {
         # with previous MIPS ISA versions, in sense that unlike
         # prior versions original MIPS binary code will fail.
         #
-        inherit_from     => [ "android64", asm("mips64_asm") ],
+        inherit_from     => [ asm("mips64_asm") ],
         perlasm_scheme   => "64",
     },
 
+    # GCC targets
+    "android" => {
+        inherit_from     => [ "android-base" ],
+        cflags           => add(picker(default => "-mandroid")),
+    },
+    "android-x86" => {
+        inherit_from     => [ "android", "android-x86-base" ],
+    },
+    "android-armeabi" => {
+        inherit_from     => [ "android", "android-armeabi-base" ],
+    },
+    "android-mips" => {
+        inherit_from     => [ "android", "android-mips-base" ],
+    },
+    "android64" => {
+        inherit_from     => [ "android64-base" ],
+        cflags           => add(picker(default => "-mandroid")),
+    },
+    "android64-aarch64" => {
+        inherit_from     => [ "android64", "android64-aarch64-base" ],
+    },
+    "android64-x86_64" => {
+        inherit_from     => [ "android64", "android64-x86_64-base" ],
+    },
+    "android64-mips64" => {
+        inherit_from     => [ "android64", "android64-mips64-base" ],
+    },
+
+    # Clang targets
+    "android-clang" => {
+        inherit_from     => [ "android-base" ],
+        cc               => "clang",
+        cxx              => "clang++",
+        cflags           => add(picker(default => "-gcc-toolchain \$(GCC_TOOLCHAIN) -fno-integrated-as")),
```

Comment:
Currently on NDK r14 beta 1, arm, aarch64 mips and x86 requires it, so I add -fno-integrated-as to those targets.
Here are failing commands in those targets: [errors.txt](https://github.com/openssl/openssl/files/719767/errors.txt). I guess x86 and mips can be fixed in NDK or LLVM. For it's fixed I suggest disabling integrated assemblers.